src/dali/ccxt_pagination.py

```diff
@@ -49,7 +50,7 @@ def __init__(
         exchange_start_time: int,
         limit: Optional[int] = None,
         markets: Optional[List[str]] = None,
-        params: Optional[Dict[str, Union[int, str, None]]] = None,
+        params: Optional[Dict[str, Union[int, str, None]]] = {},
```

Comment:
Using an empty dictionary as default value can be problematic, see: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Using `None` is safe and customary in cases like this: what was the rationale for the change?

Reply:
CCXT  needs a list for `params` or no `params`. It will throw an error if you pass `None`. So, basically, `params` should never be `None`.

Reply:
Got it: in that case you can leave the parameter default as `None` and normalize `params` at the beginning of the function:
```
params = {} if None else params
```